app/models/tax_return.rb

```diff
@@ -59,24 +61,22 @@ class TaxReturn < ApplicationRecord
   enum filing_status: { single: 1, married_filing_jointly: 2, married_filing_separately: 3, head_of_household: 4, qualifying_widow: 5 }, _prefix: :filing_status
   validates :year, presence: true
 
-  after_update_commit :send_mixpanel_status_change_event, :send_surveys
   after_update_commit { InteractionTrackingService.record_internal_interaction(client) }
 
   def state_machine
     @state_machine ||= TaxReturnStateMachine.new(self, transition_class: TaxReturnTransition)
   end
 
-  delegate :can_transition_to?, :history, :last_transition, :current_state, :last_transition_to,
+  delegate :can_transition_to?, :history, :last_transition, :last_transition_to,
            :transition_to!, :transition_to, :in_state?, :advance_to, :previous_transition, :previous_state, :last_changed_by, to: :state_machine
 
   def current_state
-    state_machine.current_state || status
+    ## backwards compatible with current implementation while TaxReturn#state can be nil
+    state || state_machine.current_state
   end
 
-  before_save do
-    if status == "prep_ready_for_prep" && status_changed?
-      self.ready_for_prep_at = DateTime.current
-    end
+  def ready_for_prep_at
+    tax_return_transitions.order(:created_at).find_by(to_state: "prep_ready_for_prep")&.created_at
```

Comment:
should we, for old records, delegate this to the old `ready_for_prep_at` column? (less important: not sure what this is used for, but maybe we can eventually get rid of the old column? not urgent though)

Reply:
now that we've looked it up and see that this is only used for the mixpanel event at the end, it's definitely not important for it to delegate to the column version

and in fact we should someday delete that column